### DISCUSSION TITLE
Display menu mobile issue

### DISCUSSION THREAD

**Discussion by jasminadimitrovska** [SnowdogApps/magento2-menu#331] [Q&A]
Hi. We are currently running Snowdog on M2 project, version 2.26.0.
We are using custom Hyva theme. 

Have one question, are there any additional steps that need to be followed for displaying the menu as part of one of the Hyva UI header (C-stacked)?

Additional issue we see, if we create a menu with the identifier: 'hyva-topmenu-mobile' and we enable the same from admin, we are seeing the following issue on frontend:
![Screenshot from 2024-07-03 16-38-11](https://github.com/SnowdogApps/magento2-menu/assets/174625759/35354704-9075-4b6c-ab1e-0ed94be088a7)
This issue is visible when 'Hyva default' theme is active as well.

**Comment by andrzejborkowski** [comment]
Hi @jasminadimitrovska 

please see the following issue https://github.com/SnowdogApps/magento2-menu/issues/306 where you can find the possible fix. 
Our team decided to maintain our menu module only on clean Hyva theme.

Regards,
Andrzej

**Reply by jasminadimitrovska** [reply]
Thank you for the updated info details Andrzej, we will double check the mentioned issue fix.

Regards,
Jasmina

**Reply by jasminadimitrovska** [reply]
Hi @andrzejborkowski 

I have tried to apply the fix from the #306 issue in my custom theme, unfortunately I'm not seeing any differences.

I have done the following two steps.

Removed the two highlighted lines in  Magento_Theme/layout/default.xml where it says:

<?xml version="1.0"?>

<page xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="urn:magento:framework:View/Layout/etc/page_configuration.xsd">
    <body>
        <move element="topmenu.mobile" destination="header-content" after="-"/>
        <move element="topmenu.desktop" destination="header-content" after="-"/>
    </body>
</page>
and in the new header file of the Hyva UI do this:
Replace <?= $block->getChildHtml(‘topmenu.mobile’) ?> to <?= $block->getChildHtml(‘topmenu’) ?>
but I do not see any difference, the mobile menu is not shown.

Other way that I've already tried is adding the following two lines in the: Magento_Theme/layout/default.xml

        <move element="topmenu-mobile" destination="header-content" before="-"/>
        <move element="topmenu-desktop" destination="header-content" after="-"/>

and then in the Magento_Theme/templates/html/header.phtml call the blocks:

         <?= $block->getChildHtml('topmenu-mobile') ?>
         <?= $block->getChildHtml('topmenu-desktop') ?>

After I see the following issue appear on frontend:
![Screenshot from 2024-07-08 12-09-56](https://github.com/SnowdogApps/magento2-menu/assets/174625759/617bcc2d-f9a0-497a-90e6-017910e14308)

**Comment by jasminadimitrovska** [comment]
Hi Andrzej,

We are using the snowdog/module-menu v.2.26.0 on our Magento 2 application.
We are using a custom hyva theme which implements the Hyva UI stacked C
header type.
In addition to the issue we encountered (Discussion #331), I have
additional questions regarding the hyva compatibility module for the
snowdog/module-menu module. Does the hyva compatibility module need to be
installed as part of the hyva-themes/magento2-snowdog-menu aside the
standard snowdog/module-menu?

Hope to hear from you soon.

Thank you in advance.

Kind regards,
Jasmina

On Wed, Jul 3, 2024 at 4:57 PM Andrzej ***@***.***> wrote:

> Hi @jasminadimitrovska <https://github.com/jasminadimitrovska>
>
> please see the following issue #306
> <https://github.com/SnowdogApps/magento2-menu/issues/306> where you can
> find the possible fix.
> Our team decided to maintain our menu module only on clean Hyva theme.
>
> Regards,
> Andrzej
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/SnowdogApps/magento2-menu/discussions/331#discussioncomment-9949462>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BJUJHX4DKFYVSSHRA2CFFT3ZKQGOBAVCNFSM6AAAAABKJXRKSKVHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM4TSNBZGQ3DE>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***
> com>
>


-- 
[image: Logo] <https://www.ecomwise.com/>
*Jasmina Poptrajkoska Dimitrovska*
Senior Magento Front End Developer
*T*: +31 (0) 30 - 22 70 465
*E*: ***@***.*** ***@***.***> | www.ecomwise.com
Reykjavikstraat 1 | 3543 KH Utrecht | The Netherlands
